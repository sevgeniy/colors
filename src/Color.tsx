import React, { FunctionComponent } from 'react';
import StarRating from './StarRating';
import './color.css';

interface IColorProps {
    title: string;
    value: string;
    rating: number;
}

const Color: FunctionComponent<IColorProps> = ({
    title,
    value,
    rating,
}): JSX.Element => {
    return (
        <div className="color">
            <h2 className="color__title">{title}</h2>
            <div
                className="color__value"
                style={{ backgroundColor: value }}
            ></div>
            <div className="color__rating">
                <StarRating starsSelected={rating} />
            </div>
        </div>
    );
};

export default Color;
