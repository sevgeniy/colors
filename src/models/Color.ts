import { v4 as uuidv4 } from 'uuid';

class Color {
    private id: string;
    private title: string;
    private value: string;
    private rating: number = 0;

    constructor(title: string, value: string, rating?: number, id?: string) {
        this.id = id === undefined ? uuidv4() : id;
        this.title = title;
        this.value = value;
        if (rating !== undefined) this.rating = rating;
    }

    get Id(): string {
        return this.id;
    }

    get Title(): string {
        return this.title;
    }

    get Value(): string {
        return this.value;
    }

    get Rating(): number {
        return this.rating;
    }
}

export default Color;
