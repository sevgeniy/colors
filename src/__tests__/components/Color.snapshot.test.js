import React from 'react';
import Color from '../../Color';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { compose } from 'redux';

configure({ adapter: new Adapter() });

const shallowExpect = compose(expect, toJson, shallow);

describe('<Color /> UI component ', () => {
    it('Renders correct properties', () => {
        shallowExpect(
            <Color
                title="red"
                value="red value"
                rating={2}
                id="9094ec55-0c1e-4b81-85d3-a5fbac0610bd"
            />
        ).toMatchSnapshot();
    });
});
