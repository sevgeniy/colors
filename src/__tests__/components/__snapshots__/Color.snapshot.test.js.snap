// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Color /> UI component  Renders correct properties 1`] = `
<div
  className="color"
>
  <button
    onClick={[Function]}
  >
    X
  </button>
  <h2
    className="color__title"
  >
    red
  </h2>
  <div
    className="color__value"
    style={
      Object {
        "backgroundColor": "red value",
      }
    }
  />
  <div
    className="color__rating"
  >
    <StarRating
      onRate={[Function]}
      starsSelected={2}
    />
  </div>
</div>
`;
